### PR TITLE
Add proper config, better readme

### PR DESCRIPTION
**Pull request by joshbuddy**

### CODE REVIEW

File: three_pass.go [hunk 1]
```diff
@@ -88,45 +122,42 @@ func InitThreePassJpakeWithCurveAndHashFns[P CurvePoint[P, S], S CurveScalar[S]]
...
+func RestoreThreePassJpakeWithCurveAndConfig[P CurvePoint[P, S], S CurveScalar[S]](userID, otherUserID, sessionKey []byte, x1, x2, s S, otherX1G, otherX2G P, curve Curve[P, S], config *Config) (*ThreePassJpake[P, S], error) {
 	jp := new(ThreePassJpake[P, S])
 	jp.userID = userID
-	jp.sessionConfirmationBytes = sessionConfirmationBytes
 	jp.OtherUserID = otherUserID
 	jp.SessionKey = sessionKey
 	jp.X1 = x1
 	jp.X2 = x2
 	jp.S = s
 	jp.OtherX1G = otherX1G
```
Comment (c53apirker) [inline three_pass.go:183]: Please validate if these points are the infinity points

File: three_pass.go [hunk 2]
```diff
@@ -88,45 +122,42 @@ func InitThreePassJpakeWithCurveAndHashFns[P CurvePoint[P, S], S CurveScalar[S]]
...
 }
 
-func RestoreThreePassJpakeWithCurveAndHashFns[P CurvePoint[P, S], S CurveScalar[S]](userID, sessionConfirmationBytes, otherUserID, sessionKey []byte, x1, x2, s S, otherX1G, otherX2G P, curve Curve[P, S], hashFn HashFnType, kdf KDFType) (*ThreePassJpake[P, S], error) {
+func RestoreThreePassJpakeWithCurveAndConfig[P CurvePoint[P, S], S CurveScalar[S]](userID, otherUserID, sessionKey []byte, x1, x2, s S, otherX1G, otherX2G P, curve Curve[P, S], config *Config) (*ThreePassJpake[P, S], error) {
 	jp := new(ThreePassJpake[P, S])
 	jp.userID = userID
-	jp.sessionConfirmationBytes = sessionConfirmationBytes
 	jp.OtherUserID = otherUserID
 	jp.SessionKey = sessionKey
 	jp.X1 = x1
```
Comment (c53apirker) [inline three_pass.go:180]: Verification that x1 and x2 are not zero would be beneficial

File: three_pass.go [hunk 3]
```diff
@@ -181,25 +221,39 @@ func (jp *ThreePassJpake[P, S]) computeZKP(x S, generator P, y P) (ZKPMsg[P, S],
...
+	// if c is zero
+	if c.BitLen() == 0 {
+		return false
+	}
+	// validate T is not 0
+	if msgObj.T.Equal(jp.curve.NewPoint()) == 1 {
+		return false
+	}
+
+	// validate public key is not at infinity
```
Comment (c53apirker) [inline three_pass.go:267]: The comment is misleading, or? R is the value r in the RFC, that is an integer mod N

File: three_pass.go [hunk 4]
```diff
@@ -181,25 +221,39 @@ func (jp *ThreePassJpake[P, S]) computeZKP(x S, generator P, y P) (ZKPMsg[P, S],
...
+	if msgObj.R.BigInt().BitLen() == 0 {
```
Comment (c53apirker) [inline three_pass.go]: Testing at least suggests that here could be a problem when R is not checked if being the curve order N

File: three_pass.go (hunk 2 above)
Comment (joshbuddy) [reply three_pass.go:180]: And s

**Pull request merged**